src/Opdex.Platform.Infrastructure/Clients/CirrusFullNodeApi/Handlers/LiquidityPools/LiquidityQuotes/CallCirrusGetLiquidityAmountInQuoteQueryHandler.cs

```diff
@@ -21,7 +21,7 @@ public CallCirrusGetLiquidityAmountInQuoteQueryHandler(ISmartContractsModule sma
         public async Task<UInt256> Handle(CallCirrusGetLiquidityAmountInQuoteQuery request, CancellationToken cancellationToken)
         {
             var quoteParams = new[] { $"12#{request.AmountA}", $"12#{request.ReserveA}", $"12#{request.ReserveB}" };
-            var localCall = new LocalCallRequestDto(request.Market, request.Market, "GetLiquidityQuote", quoteParams);
+            var localCall = new LocalCallRequestDto(request.Router, request.Router, "GetLiquidityQuote", quoteParams);
             var amountIn = await _smartContractsModule.LocalCallAsync(localCall, cancellationToken);
```

Comment:
Made these changes, then went a step further and updated the smart contract module to accept `SmartContractMethodParameter` rather than serialized strings. Then a JsonConverter is used to serialize/deserialize it. Removed `ToSmartContractParameter` now as well in favour of `SmartContractMethodParameter`.